### ISSUE TITLE
Sometimes multiple wav files are created in recorded meetings due to instability or race conditions.

### ISSUE THREAD

**Issue by mconf-daileon** [mconf/bigbluebutton#315]
---

Author Name: **Rafael Petersen** (Rafael Petersen)
Original Redmine Issue: 1962, http://dev.mconf.org/redmine/issues/1962
Original Date: 2016-04-11
Original Assignee: Felipe Cecagno

---

Some recordings will have multiple (sometimes several, as much as 40) wav files covering almost the same segment of the meeting, due to race conditions, when two people enter the voice conference and trigger the starting of the recording concurrently or due to instabilities, when the same person may happen to send multiple messages for starting the recording during lag spikes. 

We must implement safety in order to avoid this kind of issue, for it overloads the server with wrongly, unnecessary replicated data.

**Comment by mconf-daileon** [comment]
---

Original Redmine Comment
Author Name: **Rafael Petersen** (Rafael Petersen)
Original Date: 2016-04-11T17:25:03Z

---

Felipe sent me a version of the code using an extra variable to avoid the multiple start recording message and I improved the code adding syncronization, making the check and set of this variable atomic.
Race conditions should not be an issue anymore and the new variable check ensures that upon reconecting or lag spikes there is no resending of the message.
As this issue was observed in the 0.7.4 version, this implementation applies to 0.7.4 version only.

We also implemented new messages to reflect the state of the system more properly on the scala logs.

Link to the compare: https://github.com/mconf/bigbluebutton/compare/mconf-live0.7.4...RafaFP:fix-multiple-wav?diff=unified&name=fix-multiple-wav
